model/model_actor.py

```diff
@@ -0,0 +1,98 @@
+from battler import Actor, Team
+from model.ability_embedding import AbilityEmbedding
+from model.move_embedding import MoveEmbedding
+import torch
+import torch.nn as nn
+
+#
+EMBEDDING_DIM = 5
+HIDDEN_DIM = 64
+
+
+class ModelActor(Actor):
+    def __init__(self, team: Team):
+        super().__init__(team)
+
+        self.ability_embedding = AbilityEmbedding()
+        self.move_embedding = MoveEmbedding()
+
+        self.numerical_fc = nn.Linear(7 * 12, EMBEDDING_DIM)  # 7 stats for each pokemon on both teams
+
+        self.hidden_layer = nn.Sequential(
+            nn.Linear(EMBEDDING_DIM * (12 + 48 + 1), HIDDEN_DIM),
+            nn.ReLU(),
+            nn.Linear(HIDDEN_DIM, EMBEDDING_DIM)
+        )
+
+        self.action_fc = nn.Linear(EMBEDDING_DIM, 9)
+
+        self.prev_probs = None
+        self.errors = 0
+
+    def pick_move(self, knowledge) -> str:
+        # knowledge['opponent'].print_short_info()
+        # self.team.print_short_info()
+
+        # error handling - if we receive any error from our prev choice we move to the next best choice
+        # this isn't ideal because we can run into certain errors that wouldn't require making a new choice, or making a good choice
+        # ex. I've seen "invalid move - it's not your turn" in some of the logs, in which case we can keep the same move
+        # or trying to make a move with a fainted pokemon - we should immediately choose the best "switch" action
+        if knowledge['error']:
+            self.errors += 1
+            # using % 9 is a very hacky fix that needs to be removed.
```

Comment:
If errors start to wrap around, since prev_probs stays the same, why would a move or switch that previously errored not error a second time?

Reply:
It only really solves for a specific error, which is what Carter and I were talking about on Teams.

For some reason sometimes we get the feedback that it's not our turn, so our submitted move gets ignored and then skipped. So the %9 comes into play if the skipped move is our only valid move, all 8 other choices are attempted before returning to the original choice.

Reply:
In the long run we need to fix this error handling. Obviously because we don't want to iterate over 8 invalid moves and then break, but we also want to make sure in any case we're able to accurately choose the highest probability move.

Reply:
makes sense, thanks for explaining